qutebrowser/mainwindow/statusbar/bar.py

```diff
@@ -416,12 +416,9 @@ def on_mode_left(self, old_mode, new_mode):
     @pyqtSlot(browsertab.AbstractTab)
     def on_tab_changed(self, tab):
         """Notify sub-widgets when the tab has been changed."""
-        objects.statusbar_items["url"].on_tab_changed(tab)
-        objects.statusbar_items["progress"].on_tab_changed(tab)
-        # FIXME(pylbrecht): there might be only one scroll item
-        objects.statusbar_items["scroll"].on_tab_changed(tab)
-        objects.statusbar_items["scroll_raw"].on_tab_changed(tab)
-        objects.statusbar_items["history"].on_tab_changed(tab)
+        for item in objects.statusbar_items.values():
```

Comment:
Regarding the commit message:

> I am not too happy about that, because not all subclasses will implement `on_tab_changed()`

The options I see are a) this, make signals part of the `StatusBarItem` interface and make them optional to implement, b) have subclass items connect to signals they need on their own (either from the global object interfaces or from the `MainWindow` interface which could be passed down in the constructor).
There might be other options though, I haven't thought about it too much.